extension/content/components/recipes/form/arguments/MultiPreferenceBranches.js

```diff
@@ -0,0 +1,293 @@
+      updatedPref = {
+        ...prefValues,
+        [prefFieldName]: prefValue,
+        preferenceValue: "",
+      };
+    } else {
+      updatedPref = { ...prefValues, [prefFieldName]: prefValue };
+    }
+
+    /* eslint-disable no-unused-vars */
+    const {
+      [preferenceName]: _omitPrefValue,
+      ...cleanedPref
+    } = branch.preferences;
+    /* eslint-enable no-unused-vars */
+
+    const updatedPreferences = {
+      ...cleanedPref,
+      [preferenceName]: updatedPref,
+    };
+    handlePrefDataUpdate(updatedPreferences);
+
+    preferences[prefIndex] = { preferenceName, ...updatedPref };
+    setPreferences(preferences);
+  };
+
+  const handlePrefNameChange = (prefIndex, newPrefName) => {
+    const preference = preferences[prefIndex];
+    const { preferenceName: oldPrefName, ...prefValues } = preference;
+    /* eslint-disable no-unused-vars */
+    const { [oldPrefName]: _omitPrefName, ...cleanedPref } = branch.preferences;
+    /* eslint-enable no-unused-vars */
+    const updatedPref = { ...cleanedPref, [newPrefName]: prefValues };
+    handlePrefDataUpdate(updatedPref);
+    preference.preferenceName = newPrefName;
+    setPreferences(preferences);
+  };
+
+  const handlePrefDataUpdate = (updatedPref) => {
+    dispatch({
+      type: ACTION_UPDATE_DATA,
+      data: {
+        ...data,
+        arguments: {
+          ...data.arguments,
+          branches: branches.map((b, i) => {
+            if (i === index) {
+              return {
+                ...b,
+                preferences: updatedPref,
+              };
+            }
+
+            return b;
+          }),
+        },
+      },
+    });
+  };
+
+  const handleClickDelete = () => {
+    dispatch({
+      type: ACTION_UPDATE_DATA,
+      data: {
+        ...data,
+        arguments: {
+          ...data.arguments,
+          branches: branches.filter((b, i) => i !== index),
+        },
+      },
+    });
+  };
+
+  const handleChange = (name) => {
+    return (value) => {
+      dispatch({
+        type: ACTION_UPDATE_DATA,
+        data: {
+          ...data,
+          arguments: {
+            ...data.arguments,
+            branches: branches.map((b, i) => {
+              if (i === index) {
+                return { ...b, [name]: value };
+              }
+
+              return b;
+            }),
+          },
+        },
+      });
+    };
+  };
+
+  let prefList = <HelpBlock>There are no preferences.</HelpBlock>;
+  if (branch && branch.preferences) {
+    prefList = preferences.map((pref, i) => {
+      return (
+        <PreferenceFields
+          key={`${branch.slug}:${i}`}
+          index={i}
+          prefData={pref}
+          onChange={handlePrefChange}
+          onDelete={handleDeletePref}
+        />
+      );
+    });
+  }
+
+  const parseNumericInput = (value) => {
+    if (!value) {
+      return "";
+    }
+
+    return parseInt(value, 10);
+  };
+
+  return (
+    <FormGroup>
+      <div className="d-flex">
```

Comment:
I think this `div` needs to be wrapped in `FormGroup` to get the appropriate layout margin. As an aside at some point we should look at ditching all the `FormGroup`s for CSS classes that give us the margins we want.